dev/buildtool/spin_commands.py

```diff
@@ -436,6 +398,49 @@ def init_argparser(self, parser, defaults):
     BomSourceCodeManager.add_parser_args(parser, defaults)
 
 
+def bump_spin_patch(git, git_dir, gate_version):
+  '''Calculates the next spin version from the gate version and previous spin tags.
+
+  Spin is coupled to the Gate major and minor version.
+  Gate is a routing server, so features and breaking changes in Gate
+  must be reflected in spin since it is a client.
+
+  :param git: git support helper class.
+  :param git_dir: spin git directory.
+  :param gate_version: gate version to align spin version with in <maj>.<min>.<patch>-<buildnum> format.
+  :return: (SemanticVersion) Next semver spin version.
+  '''
+  gate_version_parts = gate_version.split('-')
```

Comment:
why do you need this split? Why cant you just take the semver of gate_version and ignore the "series" part and force it to "version" if you want that?

Reply:
This is actually cutting off the datestring build number. The input to this function is <maj>.<min>.<patch>-<buildnum> (specified in the docstring).

Reply:
ah, gate_version is a build version, not a source version.